src/models/spinRecord.ts

```diff
@@ -0,0 +1,32 @@
+import { Schema, Types, model } from "mongoose";
+import { ISpinRecord } from "../../types";
+
+const SpinRecordSchema: Schema = new Schema<ISpinRecord>(
+  {
+    userId: {
+      type: Types.ObjectId,
```

Comment:
UserId can't be restricted to ObjectId since different partners will use different db.